### ISSUE TITLE
BF3 chatbox lag - Possibility to turn off the chat system - Vanilla bf3 related

### ISSUE THREAD

**Issue by random444446** [VeniceUnleashed/VeniceUnleashed#448]
Would it be possible to completely turn off the chat system ?
I don't know if you have access to this core feature.

I tried to pause some bf3.exe threads with "process explorer".
Unfortunately there is not a thread for chat handling only, there is for instance one for the sound, another one for the displayed gun.
Stopping Origin.exe, web browser, even explorer.exe, playing windowed doesn't help, maybe delays a bit the issue.
I also tried to firewall the chat messages but it looks like it is in udp messages and only DPI could handle the filtering.
Setting bf3.exe affinity to cpu cores id superior to 0, unloading the first cpu core doesn't help.

It looks like chat messages handling is more important that even the video driver as there is a massive stutter with each sent / received chat message when chatbox lag appears (20-30 minutes).

Apparently same issue appeared in bc2 and has been patched, issue doesn't appear in bf4.

See this video and comments
BF3 ChatBox Lag :
https://www.youtube.com/watch?v=NSkG4IXkPHY

Less obvious video cause there is lot of cpu spikes even without chat messages in this video but in comments :
"players report it takes up to 30 minutes of server and chat spam for the chat box to cause spikes."
https://www.youtube.com/watch?v=fsIKSoTpnOA

Note : just hiding the chatbox with "h" default key doesn't help, it looks like it is the chat message processing that creates the frametimes of 60ms+  after 20-30 minutes of gameplay.
render.perfoverlayvisible 1
render.perfoverlayenable 1, that is forced to 1 after each respawn even though it is set to 0 in my user.cfg

UI.DrawEnable 0 or MaxVariableFps [screen refresh rate] doesn't help or delay a bit the issue.

**Comment by 3ti65** [comment]
Are you running Windows 10 Version 1903?

**Comment by random444446** [comment]
No, I use win 7 and will migrate to win 10 at the end of the year.
The same problem appears with win 8.1 and windows 10 October build.
Game mode etc was disabled in win 10.

The problem appears with "weak" CPUs like OCed phenom x6 or stock freq i5-4690

I get 100 fps+ for half hour without any problem.
GameTime.MaxVariableFps 75 for instance can give more game time without chatbox lag.

The moment the irreversible CPU spike for each chatbox message depends :
- on the number of chat messages per minute, server spam messages make the situation worse
- on the number of players : around 50 players can give 40 min of peaceful game

As I said in the first post, I don't know what has been reverse engineered at the moment.
So, if it is a simple task to give the possibility to turn off the chat system, and not just hide the chatbox, it would be really appreciated.

The only solution right now is to buy a whole new rig with ryzen 2600 / i7-4790k.

All this just for server spam messages.

To be clear, I'm not the only one with the chatbox lag problem.

**Comment by kiwidoggie** [comment]
At first glance this could be caused by the anti-spam written by dice. The server pushes messages to a vector and over time iterating said vector will become harder, have not tested this.

**Comment by FlashHit** [comment]
I ran some tests and found out that this happens after 200 chat messages on your client.
Also this doesn't happen when you use Advanced-Chat that replaces the vanilla chat.
Mod for repro:
[spamBot.zip](https://github.com/EmulatorNexus/VeniceUnleashed/files/5685166/spamBot.zip)
Enable the performance overlay by typing `Render.PerfOverlayVisible true` in your console.
Then just spawn and wait around 2 minutes.

From what I heard this doesn't happen on all cpu's.
I have the i7-4770k and there I have this chatlag.

**Comment by FlashHit** [comment]
Found a fix. 
Changing the [MessageQueueSize](https://github.com/EmulatorNexus/Venice-EBX/blob/b80b49f6be7010b968eca36654e70819402fa259/UI/UIComponents/UIMessageComp.txt#L36) for the chat to 6 since the chat only shows only the last 5 messages (and not 200.. wtf dice). I set it to 6 because the chatmessages getting centered after the 6th message.
Anyways here you go:
[FixChatLag.zip](https://github.com/EmulatorNexus/VeniceUnleashed/files/5685319/FixChatLag.zip)

**Comment by ghost** [comment]
FlashHit could you tell me how can i apply your method?

**Comment by FlashHit** [comment]
> 
> 
> FlashHit could you tell me how can i apply your method?

@antisept 
This is for server owners only. You as a player can't just activate the mod. Anyways most servers that run the BetterIngameAdmin also have the Chatlag Fix included. I hope this helps.

**Comment by mayco2143** [comment]
Hi I have this problem for bf3 vanilla version. Any fix found in these years? I have a 3060,5600x and 16gb ram
